### PR TITLE
Implement GitTrack handler into reconcile

### PR DESCRIPTION
**Pull request by JoelSpeed**
This PR continues the testing rewrite in the GitTrack controller started in #163

### CODE REVIEW

File: pkg/controller/gittrack/handler_test.go [hunk 1]
```diff
@@ -270,6 +281,246 @@ var _ = Describe("Handler Suite", func() {
...
+					m.Get(gto, consistentlyTimeout).Should(Succeed())
+
+					gtoList := &farosv1alpha1.GitTrackObjectList{}
+					m.Eventually(gtoList, timeout).Should(testutils.WithItems(HaveLen(3)))
+
+					By("Setting the reference to a commit after the resource was removed")
+					m.UpdateWithFunc(gt, setGitTrackReferenceFunc(repositoryURL, "28928ccaeb314b96293e18cc8889997f0f46b79b"), timeout).Should(Succeed())
+				})
+
+				It("deletes the newly removed", func() {
```
Comment (mthssdrbrg) [inline pkg/controller/gittrack/handler_test.go]:
```suggestion
				It("deletes the newly removed resource", func() {
```

File: pkg/controller/gittrack/handler_test.go [hunk 2]
```diff
@@ -270,6 +281,246 @@ var _ = Describe("Handler Suite", func() {
...
+				})
+
+				It("creates the new resources", func() {
+					m.Get(gto, timeout).Should(Succeed())
+				})
+			})
+
+			Context("and resources are removed from the repository", func() {
+				BeforeEach(func() {
+					By("Executing the handler on and older commit")
```
Comment (mthssdrbrg) [inline pkg/controller/gittrack/handler_test.go]:
```suggestion
					By("Executing the handler on an older commit")
```

**Review by mthssdrbrg — COMMENTED** [review]
There are a few contexts and its that are pending, should they be?

File: pkg/controller/gittrack/handler_test.go [hunk 3]
```diff
@@ -160,6 +159,18 @@ var _ = Describe("Handler Suite", func() {
 			})
 		}
 
+		var AssertIgnoresWrongNamespaceChild = func(r *handlerResult) {
+			It("does not create a GitTrackObject for the child", func() {
+				m.Get(gto, consistentlyTimeout).ShouldNot(Succeed())
+			})
+
+			PIt("ignores the child resource", func() {
```
Comment (JoelSpeed) [inline pkg/controller/gittrack/handler_test.go]: This test is pending as the error message has been made more verbose as part of the move to introduce `ClusterGitTrack`s

File: pkg/controller/gittrack/handler_test.go [hunk 4]
```diff
@@ -326,6 +577,55 @@ var _ = Describe("Handler Suite", func() {
 
 				AssertInvalidSubPath(kind)
 			})
+
+			PContext("when a child is owner by another controller", func() {
```
Comment (JoelSpeed) [inline pkg/controller/gittrack/handler_test.go]: The ignore logic is not yet implemented to handle the differences between `GitTrack` and `ClusterGitTrack` right now, so this must be pending

File: pkg/controller/gittrack/handler_test.go [hunk 5]
```diff
@@ -384,6 +684,26 @@ var _ = Describe("Handler Suite", func() {
...
```
Comment (JoelSpeed) [inline pkg/controller/gittrack/handler_test.go]: The ignore logic is not yet implemented to handle the differences between `GitTrack` and `ClusterGitTrack` right now, so this must be pending

**Review by mthssdrbrg — APPROVED** [review]
:+1:

**Pull request merged**